packages/ra-ui-materialui/src/list/FilterButton.js

```diff
@@ -1,122 +1,103 @@
-import React, { Component } from 'react';
+import React, { Component, useState, useCallback, useMemo } from 'react';
 import PropTypes from 'prop-types';
 import Menu from '@material-ui/core/Menu';
-import { withStyles, createStyles } from '@material-ui/core/styles';
+import { withStyles, makeStyles } from '@material-ui/core/styles';
 import ContentFilter from '@material-ui/icons/FilterList';
 import classnames from 'classnames';
 import compose from 'recompose/compose';
-import { translate } from 'ra-core';
 import lodashGet from 'lodash/get';
 
 import FilterButtonMenuItem from './FilterButtonMenuItem';
 import Button from '../button/Button';
 
-const styles = createStyles({
+const useStyles = makeStyles({
     root: { display: 'inline-block' },
 });
 
-export class FilterButton extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            open: false,
-            anchorEl: null,
-        };
-        this.handleClickButton = this.handleClickButton.bind(this);
-        this.handleRequestClose = this.handleRequestClose.bind(this);
-        this.handleShow = this.handleShow.bind(this);
-    }
+const FilterButton = ({
+    showFilter,
+    filters,
+    displayedFilters,
+    filterValues,
+    classes = {},
+    className,
+    resource,
+    ...rest
+}) => {
+    const [open, setOpen] = useState(false);
+    const [anchorEl, setAncholEl] = useState();
+    const styles = useStyles();
```

Comment:
```suggestion
    const styles = useStyles({ classes: classesOverride });
```
Allows users to override classes